src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -96,56 +127,81 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   }
 
   /**
-   * Uses the Google Cloud Vision API to find faces in the image represented by the binary data
-   * stored in @param imageBytes
+   * Uses the Google Cloud Vision API to find parts to blur in the image represented by the binary
+   * data stored in @param imageBytes
    *
-   * @return an ArrayList of bounding rectangles representing faces. Rectangles are represented by a
-   *     list of points. There is no guaranteed order of the points.
+   * @return an ArrayList of bounding rectangles representing parts to blur. Rectangles are
+   *     represented by a list of points. There is no guaranteed order of the points.
    */
-  private ArrayList<List<Point>> getBlurAreas(byte[] imageBytes) throws IOException {
+  private ArrayList<List<Point>> getBlurAreas(
+      byte[] imageBytes, Integer partsToBlurMask) throws IOException {
     // This is the array that we will return.
-    ArrayList<List<Point>> faceBorders = new ArrayList<>();
+    ArrayList<List<Point>> rectanglesToBlur = new ArrayList<>();
 
     // Convert bytes to an Image object.
     ByteString byteString = ByteString.copyFrom(imageBytes);
     Image image = Image.newBuilder().setContent(byteString).build();
 
-    // Create a face detection request to our image.
-    Feature feature = Feature.newBuilder().setType(Feature.Type.FACE_DETECTION).build();
-    AnnotateImageRequest request =
-        AnnotateImageRequest.newBuilder().addFeatures(feature).setImage(image).build();
-
-    // Create an array of requests containing only ours to pass to the batchAnnotateImages function.
+    // Create an array of requests containing our features for the batchAnnotateImages function.
     List<AnnotateImageRequest> requests = new ArrayList<>();
-    requests.add(request);
+
+    // Check which parts our bitmask contains and add the corresponding features to requests array.
+    if((partsToBlurMask & FACE_BLUR_MASK) != 0) {
+      requests.add(createAnnotateImageRequest(image, Feature.Type.FACE_DETECTION));
+    }
+    if((partsToBlurMask & PLATE_BLUR_MASK) != 0) {
+      requests.add(createAnnotateImageRequest(image, Feature.Type.OBJECT_LOCALIZATION));
+    }
+    if((partsToBlurMask & LOGO_BLUR_MASK) != 0) {
+      requests.add(createAnnotateImageRequest(image, Feature.Type.LOGO_DETECTION));
+    }
 
     // Annotate the image from our request. Skip if there is any internal error in the API.
     try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
       BatchAnnotateImagesResponse response = client.batchAnnotateImages(requests);
       List<AnnotateImageResponse> responses = response.getResponsesList();
```

Comment:
Is it like responses[i] is a response to requests[i] request? It means that one response contain only one type of annotation, it can not contain both car plates and faces, for example, right?

Reply:
I'm also interested in this.

Reply:
A response corresponds to a request, right. But a request can contain both car plates and faces now. (see first review comment)